compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -962,16 +967,67 @@ static void getDefaultMatmulVectorSizes(
   return;
 }
 
+/// Checks if the input and output types of a linalg op are the same, and if so
+/// returns the type. Otherwise, returns failure.
+static FailureOr<Type> nonWideningLinalgElementType(linalg::LinalgOp op) {
+  SmallVector<Type, 3> inputAndOutputElementTypes;
+  for (Value v :
+       llvm::concat<Value>(op.getRegionInputArgs(), op.getRegionOutputArgs())) {
+    inputAndOutputElementTypes.push_back(v.getType());
+  }
+  if (inputAndOutputElementTypes.empty() ||
```

Comment:
Can we remove the `.empty()` check? I think this is an invalid linalg op if it happens.

Reply:
I've changed this to a debug assert :+1: